numba/cuda/target.py

```diff
@@ -143,6 +144,8 @@ def prepare_cuda_kernel(self, codelib, func_name, argtypes, debug,
         argtypes:      An iterable of the types of arguments to the kernel.
         debug:         Whether to compile with debug.
         nvvm_options:  Dict of NVVM options used when compiling the new library.
+        filename:      The source filename that the function is contained in.
+        linenum:       The source line that the function is on.
```

Comment:
What happens if someone materialized a function from a string?

Reply:
`filename` is `<string>` and `linenum` is the line in the string on which the function began - I think this is normal / expected?